curp/src/server/raw_curp/mod.rs

```diff
@@ -310,6 +313,16 @@ impl<C: 'static + Command, RC: RoleChange + 'static> RawCurp<C, RC> {
                 Err(ConfChangeError::new_propose(ProposeError::NotLeader)),
             );
         }
+        if let Some(change) = conf_change.changes().first() {
+            if let ConfChangeType::Promote = change.change_type() {
+                let learner_index = self.lst.get_match_index(change.node_id);
+                let leader_index = self.log.read().last_log_index();
+                if leader_index.overflow_sub(learner_index) > MAX_PROMOTE_GAP {
+                    return (info, Err(ConfChangeError::LearnerNotCatchUp(())));
+                }
```

Comment:
Should the learner be automatically promoted? Or can the user force promote a learner without satisfying the gap?

Reply:
Currently, we need to do promote manually. In some scenarios, the user may want to keep the node in the learner state as a read-only or standby node. and learner should.
For the second question, We can add this feature, but why do we need it? if user want to add a new node and do not care about this gap, they and directly add a voter node.

Reply:
makes sense, thx.